src/main/java/org/sigmah/shared/command/result/Authentication.java

```diff
@@ -217,8 +222,31 @@ public String toString() {
 		builder.append("organizationId", organizationId);
 		builder.append("organizationName", organizationName);
 		builder.append("organizationLogo", organizationLogo);
-		builder.append("orgUnitId", orgUnitId);
-		builder.append("aggregatedProfile", aggregatedProfile);
+		builder.append("mainOrgUnitId", mainOrgUnitId);
+
+		StringBuilder serializedSecondaryOrgUnitIds = new StringBuilder("[");
+		for (Iterator<Integer> iterator = secondaryOrgUnitIds.iterator(); iterator.hasNext();) {
+			Integer secondaryOrgUnitId = iterator.next();
+			serializedSecondaryOrgUnitIds.append(secondaryOrgUnitId);
+			if (iterator.hasNext()) {
+				serializedSecondaryOrgUnitIds.append(", ");
+			}
+		}
+		serializedSecondaryOrgUnitIds.append("]");
+
+		builder.append("secondaryOrgUnitIds", serializedSecondaryOrgUnitIds.toString());
+
+		StringBuilder serializedAggregatedProfiles = new StringBuilder("[");
+		for (Iterator<Map.Entry<Integer, ProfileDTO>> iterator = aggregatedProfiles.entrySet().iterator(); iterator.hasNext(); ) {
+			Map.Entry<Integer, ProfileDTO> entry = iterator.next();
+			serializedAggregatedProfiles.append("{" + entry.getKey() + ": " + entry.getValue() + "}");
+			if (iterator.hasNext()) {
+				serializedAggregatedProfiles.append(", ");
+			}
+		}
+		serializedAggregatedProfiles.append("]");
+		builder.append("aggregatedProfiles", serializedAggregatedProfiles);
```

Comment:
This code is quite understandable but if this code of construction occur elsewhere, please introduce a utility class or just use something like Guava's com.google.common.base.Joiner

```java
Joiner.on(", ").withKeyValueSeparator(": ").join(aggregatedProfiles);
```